mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/mesos/VirualMachineWorkerServiceMesosImpl.java

```diff
@@ -63,33 +58,27 @@ public void start() {
         mesosDriver = new MesosExecutorDriver(new MesosExecutorCallbackHandler(executeStageRequestObserver));
         // launch driver on background thread
         logger.info("launch driver on background thread");
-        executor.execute(new Runnable() {
-            @Override
-            public void run() {
-                try {
-                    mesosDriver.run();
-                } catch (Exception e) {
-                    logger.error("Failed to register Mantis Worker with Mesos executor callbacks", e);
-                }
+        executor.execute(() -> {
+            try {
+                mesosDriver.run();
+            } catch (Exception e) {
+                logger.error("Failed to register Mantis Worker with Mesos executor callbacks", e);
             }
         });
         // subscribe to vm task updates on current thread
         logger.info("subscribe to vm task updates on current thread");
-        vmTaskStatusObservable.subscribe(new Action1<VirtualMachineTaskStatus>() {
-            @Override
-            public void call(VirtualMachineTaskStatus vmTaskStatus) {
-                TYPE type = vmTaskStatus.getType();
-                if (type == TYPE.COMPLETED) {
-                    Protos.Status status = mesosDriver.sendStatusUpdate(TaskStatus.newBuilder()
-                            .setTaskId(TaskID.newBuilder().setValue(vmTaskStatus.getTaskId()).build())
-                            .setState(TaskState.TASK_FINISHED).build());
-                    logger.info("Sent COMPLETED state to mesos, driver status=" + status);
-                } else if (type == TYPE.STARTED) {
-                    Protos.Status status = mesosDriver.sendStatusUpdate(TaskStatus.newBuilder()
-                            .setTaskId(TaskID.newBuilder().setValue(vmTaskStatus.getTaskId()).build())
-                            .setState(TaskState.TASK_RUNNING).build());
-                    logger.info("Sent RUNNING state to mesos, driver status=" + status);
-                }
+        vmTaskStatusObservable.subscribe(vmTaskStatus -> {
+            TYPE type = vmTaskStatus.getType();
+            if (type == TYPE.COMPLETED) {
```

Comment:
use switch-case?

Reply:
Fixed in 4526f9258f349741fdaed90f1472b98ff9628d4c